assets/tests/hooks/useDetour.test.ts

```diff
@@ -28,7 +33,11 @@ describe("useDetour", () => {
     const start = { lat: 0, lon: 0 }
     const end = { lat: 1, lon: 1 }
 
-    const { result } = renderHook(useDetour)
+    const { result } = renderHook(useDetourWithFakeRoutePattern)
+
+    jest
+      .mocked(fetchDetourMissedStops)
+      .mockImplementation(() => instantPromise(null))
```

Comment:
suggestion: Can this be changed to use `mockResolvedValue` and `waitFor` like the other tests?
I really don't like `instantPromise`, especially when we have `mockResolvedValue` and `Promise.resolve`, as this sidesteps some of the react rendering things (I'll detail that more later).

Reply:
I also want to avoid `instantPromise`, but I've been having a really hard time getting `mockResolvedValue` to work for many of these tests, especially if there isn't a change in the ultimate result of the hook that I can `waitFor`. Maybe I'm missing something, though.